partialWaveFit/fitResult.cc

```diff
@@ -569,14 +569,17 @@ double
 fitResult::intensityErr(const string& waveNamePattern) const
 {
 	// get amplitudes that correspond to wave name pattern
-	const vector<unsigned int> prodAmpIndices = prodAmpIndicesMatchingPattern(waveNamePattern);
-	const unsigned int         nmbAmps        = prodAmpIndices.size();
-	if (not _covMatrixValid or (nmbAmps == 0)) {
+	const vector<unsigned int> waveIndices    = waveIndicesMatchingPattern(waveNamePattern);
+	const unsigned int         nmbWaves       = waveIndices.size();
+	if (not _covMatrixValid or (nmbWaves == 0)) {
 		return 0.;
 	}
-	if(nmbAmps == 1) {
-		return intensityErr(prodAmpIndices[0]);
+	if(nmbWaves == 1) {
+		return intensityErr(waveIndices[0]);
 	}
+
+	const vector<unsigned int> prodAmpIndices = prodAmpIndicesMatchingPattern(waveNamePattern);
```

Comment:
As far as I understand the code, 'prodAmpIndicesForWave()' resolves the amplitude ids by the corresponding wave name, which is of course computationally quite expensive. 
However, when calling an element of the spin-density matrix, the 'prodAmpIndicesForWave()' is called anyway to resolve the mapping between amplitude ids and wave ids. 
Is there no direct mapping between the amplitude ids and the wave id, because I thought for rank=1 fits they are the same. 
Or would it be possible / sensible to construct such a mapping once and store it with the fit result?